common/src/main/java/com/revenuecat/purchases/common/CustomerInfoResponseHandler.kt

```diff
@@ -0,0 +1,96 @@
+package com.revenuecat.purchases.common
+
+import com.google.gson.internal.bind.util.ISO8601Utils
+import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.ProductType
+import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.purchases.VerificationResult
+import com.revenuecat.purchases.models.PurchasedProduct
+import com.revenuecat.purchases.utils.Iso8601Utils
+import org.json.JSONObject
+import java.util.Date
+
+class CustomerInfoResponseHandler(
+    private val billing: BillingAbstract,
+    private val dateProvider: DateProvider = DefaultDateProvider()
+) {
+
+    fun computeOfflineCustomerInfo(
+        appUserID: String,
+        onSuccess: (CustomerInfo) -> Unit,
+        onError: (PurchasesError) -> Unit
+    ) {
+        billing.queryPurchasedProducts(
+            appUserID,
+            onSuccess = { purchasedProducts ->
+                val jsonObject = JSONObject()
+                val requestDate = dateProvider.now
+                jsonObject.apply {
+                    val formattedDate = Iso8601Utils.format(requestDate)
+                    put("request_date", formattedDate)
+                    put("request_date_ms", requestDate.time)
+                    put("subscriber", JSONObject().apply {
+                        put("original_app_user_id", appUserID)
+                         put("original_application_version", null) // TODO: I think this is fine
+                        put("entitlements", generateEntitlementsResponse(purchasedProducts))
+                        put("first_seen", formattedDate)
+                        val originalPurchaseDate = calculateOriginalPurchaseDate(purchasedProducts)
+                        put("original_purchase_date", originalPurchaseDate)
+                        put("non_subscriptions", JSONObject()) // TODO
```

Comment:
we'll be able to tell these apart from subs but since we won't have revenuecat transaction ids, we can't show them, so we might as well treat them like subscriptions like the load shedder does

Reply:
Done here https://github.com/RevenueCat/purchases-android/pull/958